### PR TITLE
Prepend page path when tracking anchor links in GA4

### PR DESCRIPTION
**Pull request by AshGDS**
## What

- Ensures the page path is added to anchor links tracked by the `ga4-link-tracker` and `ga4-event-tracker`
- For example, `#england-and-wales` will become `/bank-holidays#england-and-wales` in the GA4 dashboard.
- I also cleaned up the link tracker slightly as we kept repeating `window.GOVUK.analyticsGa4.core.trackFunctions`, let me know your thoughts.
- Will add CHANGELOG when the pending release PR is live

## Why

https://trello.com/c/7thzJTbm/672-include-page-path-in-anchor-links

## Visual Changes

None.

### CODE REVIEW

File: app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js [hunk 1]
```diff
@@ -205,6 +205,10 @@ window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
         }
       },
 
+      getPathname: function () {
+        return window.location.pathname
+      },
+
```
Comment (AshGDS) [inline app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js:211]: This is wrapped in a function so that it can be mocked in testing

File: app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js [hunk 2]
```diff
@@ -62,13 +62,13 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
 
   Ga4LinkTracker.prototype.trackClick = function (event) {
     var element = event.target
-
+    var trackFunctions = window.GOVUK.analyticsGa4.core.trackFunctions
```
Comment (andysellick) [inline app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js:65]: I'm in two minds about this. It does make the code cleaner and easier to read, but at the same time it slightly obfuscates what is being called by this variable - you have to read this bit of the code to know what `trackFunctions` refers to (although it is named the same thing, so that's good).

It might make a difference to the asset size on the public site, but only fractionally, and it's possible any saving would be negated by gzip/minifying anyway.

In conclusion... I don't know! It's probably fine 😁

**Review by andysellick — APPROVED** [review]
Seems good. Don't forget to include a changelog entry.

File: app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js (hunk 2 above)
Comment (AshGDS) [reply app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js:65]: Thanks @andysellick - I've asked another dev and they liked the refactor, so will keep it for now :+1:
